pages/api/v1/answers.ts

```diff
@@ -747,18 +757,25 @@ async function Angela( // Good ol' Angela
 
       // Below checks for if there are any 'needs correction' messages
       if (commandMapOutput.some((output) => output === null)) {
-        const correctionMessage = {
-          role: "assistant",
-          content:
-            "<<[NEW-MESSAGE]>>" +
-            "Tell user:\n" +
-            "I'm sorry but I need more information before I can do that, please can you provide: " +
-            toUserCorrect.join("\n"),
-        } as ChatGPTMessage;
-
-        nonSystemMessages.push(correctionMessage);
-        streamInfo(correctionMessage);
-        return { nonSystemMessages, cost: totalCost, numUserQueries };
+        const funcToArrToCorrect: { [param: string]: string[] } = {};
+        toUserCorrect.map((toCorrect) => {
+          if (toCorrect.functionName in funcToArrToCorrect) {
+            funcToArrToCorrect[toCorrect.functionName].push(toCorrect.param);
+          } else {
+            funcToArrToCorrect[toCorrect.functionName] = [toCorrect.param];
+          }
+        });
+        Object.entries(funcToArrToCorrect).map(([functionName, params]) => {
+          const missingParamsMessage = {
```

Comment:
Can you add some comments here / clarify the variable names? I don' understand `funcToArrToCorrect` :(

Reply:
Added comments